challenges/product/challenge-1.md

```diff
@@ -0,0 +1,156 @@
+# Product Challenge
+
+## Rationale
+
+This exercise has two goals:
+
+It helps us understand what to expect from you as a technical leader, and
+whether you have the technical aptitude to engage with highly technical
+customers. It gives you a feel for what it’s like to work at Teleport. The
+exercise simulates day-to-day work, helping you better understand the team and
+our hiring process.
+
+We believe this approach is not only more effective, but also more enjoyable
+than the whiteboard or quiz-style interviews common in the industry.
+
+We appreciate your time and look forward to hacking on this little project
+together.
+
+
+## Objective
+
+Create a design document and implement a service that will let users sign on
+through SSO IDP of of any kind and get a WireGuard public key set up on their
+computer to be authorized to access a WireGuard network.
+
+## Requirements
+
+Create automation using Terraform, helm charts, or docker-compose that sets up a
+demo environment with the following capabilities:
+
+- Determine implementation, security and scope requirements for the document.
+- Present the design to the interview team - a mix of technical and business audience, explain value proposition and answer technical questions to both.
+- Explain the business and technical value of the proposed solution.
+
+Here is an example command-line tool `wg` that will let users sign in with SSO.
+After a succesful login users get a WireGuard key registered in the provisioned
+WireGuard newtork.
+
+```
+$ wg login
+# ...opens browser with login page
+# ...on successful login
+You are logged in. WireGuard is activated.
+```
+
+## Guidance
+
+### Interview Process
+
+You will join a Slack channel with the interview team. The team consists of the
+peers who will be working with you.
+
+Ask them about the culture, work and life balance, or anything else that you
+would like to learn about Teleport.
+
+Before writing the actual code, create a brief design document in Google Docs or
+markdown in GitHub and share with the team.
+
+This document should consist of key trade-offs and key design approaches. Please
+avoid writing an overly detailed design document. Use this document to make sure
+the team could provide feedback on your design and demonstrate that you've
+investigated the problem space.
+
+After the submission, prepare the presentation and a demo and schedule a 45
+minute demo call with the interview team, present your solution to the audience
+and answer any questions.
+
+After the demo, the interview team will assemble and vote using +1, -2 anonymous
+voting system: +1 is submitted whenever a team member accepts the submission, -2
+otherwise.
+
+In case of a positive result, we will connect you to our HR team who will
+collect one-two references and will work out other details. You can start the
+reference collection process in parallel if you would like to speed up the
+process.
```

Comment:
Should we remove this? Are we actually collecting references before the recruiting team reaches out with an offer?

Reply:
For this position, yes, I would like to collect some references.